polaris-icons/package.json

```
@@ -63,5 +63,8 @@
     "js-yaml": "^4.0.0",
     "rehype-parse": "^7.0.1",
     "unified": "^9.2.1"
+  },
+  "peerDependencies": {
+    "react": "*"
   }
```

Some consumers of `@shopify/polaris-icons` might not use React and instead use the raw SVG assets. Should we make this an [optional peer dependency](https://pnpm.io/next/package_json#peerdependenciesmetaoptional)?

```suggestion
  },
  "peerDependenciesMeta": {
    "react": {
      "optional": true
    }
  }
```

---

I think `peerDependenciesMeta.optional` is more intended for when your library can fallback if a peer dependency isn't available, this one will fail outright if this part of the package is used without React being available, so it seems like it probably should be an error on install if they don't have React. It might mean that the more technically correct thing would be to have separate`@shopify/polaris-icons` and `@shopify/polaris-icons-react` pacakges.

That said, we can still add this in if we prefer the warning be suppressed

---

I think you're both correct here.

To preserve backwards compatibility (even though it's technically wrong), I suggest adding the `peerDependenciesMeta.optional` flag in this PR, then in a follow up issue we can remove the optionality and ship that with the next major release (introducing an error during installation counts as a breaking change IMO)

---

Good suggestion @jesstelford and thanks for updating this @ryanwilsonperkin. That strategy sounds good to me